data_types/softlayer_dns_domain.go

```diff
@@ -0,0 +1,20 @@
+package data_types
+
+type SoftLayer_Dns_Domain_Template struct {
+	Name            string                                 `json:"name"`
+	ResourceRecords []SoftLayer_Dns_Domain_Resource_Record `json:"resourceRecords"`
+}
+
+type SoftLayer_Dns_Domain_Template_Parameters struct {
+	Parameters []SoftLayer_Dns_Domain_Template `json:"parameters"`
+}
+
+type SoftLayer_Dns_Domain struct {
+	Id         int    `json:"id"`
```

Comment:
That shouldn't be a problem, I'll make the changes and test on my side tomorrow.